### PR TITLE
Logging tweaks for labextension update

### PR DESCRIPTION
**Pull request by vidartf**
## References

Follow up to #6711.

## Code changes

- Logs a warning if no compatible version could be found when updating an extension (warnings goes to stderr by default, so the user will see it).
- Logs to debug if a deprecated, compatible version was found when updating an extension. This way, if the user cannot find a compatible version, the `--debug` flag will give the extra details to troubleshoot an issue,

## User-facing changes

When updating a package with no compatible versions:

```
jupyter labextension update @jupyter-widgets/jupyterlab-manager
No compatible version found for @jupyter-widgets/jupyterlab-manager
```

When adding the `--debug` flag:

```
jupyter labextension update @jupyter-widgets/jupyterlab-manager --debug
<debug output of paths snipped>
Node v11.9.0

Fetching URL: https://registry.npmjs.org/@jupyter-widgets%2Fjupyterlab-manager
Disregarding compatible version of package as it is deprecated: @jupyter-widgets/jupyterlab-manager@0.39.1
Disregarding compatible version of package as it is deprecated: @jupyter-widgets/jupyterlab-manager@0.39.0
Disregarding compatible version of package as it is deprecated: @jupyter-widgets/jupyterlab-manager@0.39.0-alpha.0
No compatible version found for @jupyter-widgets/jupyterlab-manager
```

## Backwards-incompatible changes

None.

### CODE REVIEW

File: jupyterlab/commands.py [hunk 1]
```diff
@@ -1475,14 +1476,17 @@ def sort_key(key_value):
...
                                     reverse=True):
-            # skip deprecated versions
-            if 'deprecated' in data:
-                continue
-
             deps = data.get('dependencies', {})
             errors = _validate_compatibility(name, deps, core_data)
             if not errors:
                 # Found a compatible version
+                # skip deprecated versions
```
Comment (jasongrout) [inline jupyterlab/commands.py]: Should we also do this in the other place we skip deprecated packages? I'm not sure when that is being used vs this one.

Comment (vidartf) [reply jupyterlab/commands.py]: I hid a little note in the commit message:
> Only log in `_latest_compatible_package_version`, not `latest_compatible_package_versions`, due to the difference in use: `_latest_compatible_package_version` is only called from install/update commands.

Comment (vidartf) [reply jupyterlab/commands.py]: The other version is typically called by the extension manager, which is trying to check if any of the installed extensions are outdated (if so, display an "Update" button).

**Comment by jasongrout** [comment]
Thanks!

**Pull request merged**